pkgs/by-name/ta/tailwindcss_4/package.nix

```
@@ -0,0 +1,76 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  runCommand,
+  autoPatchelfHook,
+  makeWrapper,
+  tailwindcss_4,
+}:
+let
+  version = "4.0.6";
+  inherit (stdenv.hostPlatform) system;
+  throwSystem = throw "tailwindcss has not been packaged for ${system} yet.";
+
+  plat =
+    {
+      aarch64-darwin = "macos-arm64";
+      aarch64-linux = "linux-arm64";
+      x86_64-darwin = "macos-x64";
+      x86_64-linux = "linux-x64";
+    }
+    .${system} or throwSystem;
+
+  hash =
+    {
+      aarch64-darwin = "sha256-FopnfuGYBjjQkfPHdIWacpom9WsAVRwHPQ/14waWR7s=";
+      aarch64-linux = "sha256-y+9d53skmoSZPhbtk9vUkhKNzIE1A868Lnkw7Cot/PU=";
+      x86_64-darwin = "sha256-rFjaDtPjJO9W6G//7Uu5CJE+pB0sqLrMByC5pv4FA4E=";
+      x86_64-linux = "sha256-14EMXenpvPsKlnu5kebNu3MmfWwa7UYfAgEkh5EwVRM=";
+    }
+    .${system} or throwSystem;
+in
+stdenv.mkDerivation {
+  inherit version;
+  pname = "tailwindcss_4";
+
+  src = fetchurl {
+    url =
+      "https://github.com/tailwindlabs/tailwindcss/releases/download/v${version}/tailwindcss-" + plat;
+    inherit hash;
+  };
+
+  nativeBuildInputs = lib.optional stdenv.hostPlatform.isLinux autoPatchelfHook;
+  buildInputs = [ makeWrapper ];
+
+  dontUnpack = true;
+  dontBuild = true;
+  dontStrip = true;
+
+  installPhase = ''
+    mkdir -p $out/bin
+    install -m755 $src $out/bin/tailwindcss
+  '';
+
+  # libstdc++.so.6 for @parcel/watcher
+  postFixup = ''
+    wrapProgram $out/bin/tailwindcss --prefix LD_LIBRARY_PATH : ${
+      lib.makeLibraryPath [ stdenv.cc.cc.lib ]
+    }
+  '';
+
+  passthru.tests.helptext = runCommand "tailwindcss-test-helptext" { } ''
+    ${tailwindcss_4}/bin/tailwindcss --help > $out
+  '';
```

```suggestion
  nativeInstallCheckInputs = [ versionCheckHook ];
  doInstallCheck = true;
  versionCheckProgramArg = "--help";
```
I think using `versionCheckHook` is a better test in this case.

---

Thanks!